### ISSUE TITLE
[terraform] Possible for provider to depend on modules?

### ISSUE THREAD

**Issue by boxcee** [mrparkers/terraform-provider-keycloak#516]
So, not sure what to do about this. Trying to import a realm into keycloak after provisioning keycloak using a helm chart.

In our code we do this:
```
module "gke_auth" {
  source       = "terraform-google-modules/kubernetes-engine/google//modules/auth"
  version      = "~> 14.0.1"
  project_id   = var.gcp_project_name
  location     = var.gcp_region
  cluster_name = var.cluster_name
}

provider "kubernetes" {
  load_config_file       = false
  cluster_ca_certificate = module.gke_auth.cluster_ca_certificate
  host                   = module.gke_auth.host
  token                  = module.gke_auth.token
}
```
And I was hoping to do the same with the `keycloak` provider, but it doesn't work:
```
module "keycloak_workload" {
  ...
}

provider "keycloak" {
  client_id = "admin-cli"
  username  = module.keycloak_workload.keycloak_admin_username
  password  = module.keycloak_workload.keycloak_admin_password
  url       = module.keycloak_workload.keycloak_hostname
}
```
Unfortunately this doesn't work and the `keycloak` provider complains about needing a `username` and `password` specified.

Is this something that could be achieved?

**Comment by mrparkers** [comment]
Hi @boxcee - could you give this a try with version `3.2.0-rc.0`?  I just published it today to test the `darwin_arm64` build, but it also includes a fix for this issue.

**Comment by mrparkers** [comment]
This should be fixed in v3.2.0, so I'm going to close this.  Please re-open it if you're still having issues.
